src/probability/kernel/cond_cdf.lean

```diff
@@ -0,0 +1,948 @@
+  (hf : ∀ b, measurable (f b)) (hf_int : ∀ b, ∫⁻ a, f b a ∂μ ≠ ∞) (h_directed : directed (≥) f) :
+  ∫⁻ a, ⨅ b, f b a ∂μ = ⨅ b, ∫⁻ a, f b a ∂μ :=
+begin
+  casesI nonempty_encodable β,
+  casesI is_empty_or_nonempty β,
+  { simp only [with_top.cinfi_empty, lintegral_const],
+    rw [ennreal.top_mul, if_neg],
+    simp only [measure.measure_univ_eq_zero, hμ, not_false_iff], },
+  inhabit β,
+  have : ∀ a, (⨅ b, f b a) = (⨅ n, f (h_directed.sequence f n) a),
+  { refine λ a, le_antisymm (le_infi (λ n, infi_le _ _))
+      (le_infi (λ b, infi_le_of_le (encodable.encode b + 1) _)),
+    exact (h_directed.sequence_le b a), },
+  calc ∫⁻ a, ⨅ b, f b a ∂μ
+        = ∫⁻ a, ⨅ n, f (h_directed.sequence f n) a ∂μ : by simp only [this]
+    ... = ⨅ n, ∫⁻ a, f (h_directed.sequence f n) a ∂μ :
+      by { rw lintegral_infi (λ n, _) h_directed.sequence_anti,
+        { exact hf_int _, },
+        { exact hf _, }, }
+    ... = ⨅ b, ∫⁻ a, f b a ∂μ :
+    begin
+      refine le_antisymm (le_infi (λ b, _)) (le_infi (λ n, _)),
+      { exact infi_le_of_le (encodable.encode b + 1) (lintegral_mono $ h_directed.sequence_le b) },
+      { exact infi_le (λb, ∫⁻ a, f b a ∂μ) _ },
+    end
+end
+
+-- todo: move to measure_theory/pi_system
+lemma is_pi_system_Iic [semilattice_inf α] : @is_pi_system α (range Iic) :=
+by { rintros s ⟨us, rfl⟩ t ⟨ut, rfl⟩ _, rw [Iic_inter_Iic], exact ⟨us ⊓ ut, rfl⟩, }
+
+-- todo: move to measure_theory/pi_system
+lemma is_pi_system_Ici [semilattice_sup α] : @is_pi_system α (range Ici) :=
+by { rintros s ⟨us, rfl⟩ t ⟨ut, rfl⟩ _, rw [Ici_inter_Ici], exact ⟨us ⊔ ut, rfl⟩, }
+
+
+end aux_lemmas_to_be_moved
+
+namespace measure_theory.measure
+
+variables {α β : Type*} {mα : measurable_space α} (ρ : measure (α × ℝ))
+
+include mα
+
+/-- Measure on `α` such that for a measurable set `s`, `ρ.Iic_snd r s = ρ (s ×ˢ Iic r)`. -/
+noncomputable
+def Iic_snd (r : ℝ) : measure α := (ρ.restrict (univ ×ˢ Iic r)).fst
+
+lemma Iic_snd_apply (r : ℝ) {s : set α} (hs : measurable_set s) :
+  ρ.Iic_snd r s = ρ (s ×ˢ Iic r) :=
+by rw [Iic_snd, fst_apply hs,
+  restrict_apply' (measurable_set.univ.prod (measurable_set_Iic : measurable_set (Iic r))),
+  ← prod_univ, prod_inter_prod, inter_univ, univ_inter]
+
+lemma Iic_snd_univ (r : ℝ) : ρ.Iic_snd r univ = ρ (univ ×ˢ Iic r) :=
+Iic_snd_apply ρ r measurable_set.univ
+
+lemma Iic_snd_mono {r r' : ℝ} (h_le : r ≤ r') : ρ.Iic_snd r ≤ ρ.Iic_snd r' :=
+begin
+  intros s hs,
+  simp_rw Iic_snd_apply ρ _ hs,
+  refine measure_mono (prod_subset_prod_iff.mpr (or.inl ⟨subset_rfl, Iic_subset_Iic.mpr _⟩)),
+  exact_mod_cast h_le,
+end
+
+lemma Iic_snd_le_fst (r : ℝ) : ρ.Iic_snd r ≤ ρ.fst :=
+begin
+  intros s hs,
+  simp_rw [fst_apply hs, Iic_snd_apply ρ r hs],
+  exact measure_mono (prod_subset_preimage_fst _ _),
+end
+
+lemma Iic_snd_ac_fst (r : ℝ) : ρ.Iic_snd r ≪ ρ.fst :=
+measure.absolutely_continuous_of_le (Iic_snd_le_fst ρ r)
+
+instance {ρ : measure (α × ℝ)} [is_finite_measure ρ] (r : ℝ) : is_finite_measure (ρ.Iic_snd r) :=
```

Comment:
If this will only ever be used in this file, I wouldn't declare it as a global instance. Just make it a lemma and add it a local instance attribute.